src/logger/httpLoggerRules_test.go

```diff
@@ -93,13 +95,104 @@ func TestUsesAllowHttpUrlRules(t *testing.T) {
 
 // test uses copy session field rules test
 
-// func TestUsesCopySessionFieldRules(t *testing.T) {
-
-// }
+func TestUsesCopySessionFieldRules(t *testing.T) {
+	// helper for function tests
+	helper := NewTestHelper()
+	// requests used for all tests in function
+	request := NewMockRequestWithJson2()
+	request.Session().SetAttribute("butterfly", "poison") //getting session and setting an attribute of it
```

Comment:
the java logger is using methods specific to `HttpServletRequest`s we need to use methods [see here](https://golang.org/pkg/net/http/#Request) probably going to use Cookie methods

Reply:
Sounds good, I will probably follow up with you after I sort this out today.